Emby.Naming/Video/DateTimeResolvers/LatestPlausibleDateMovieDateTimeResolver.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Emby.Naming.Common;
+
+namespace Emby.Naming.Video.DateTimeResolvers;
+
+/// <summary>
+/// There may be multiple dates in the file but not all of them are plausible as productionDates
+/// If there is only one Plausible one we can safely assume this is the actual date
+/// If there are multiple plausible ones (before today) we can just try to pick the latest one - I mean when the hell was a movie produced in eg 1968 calling it 2001 or so and it being the future?
+///        (except '2001: A Space Odyssey - 1968' thats a edge case and if they dont bracket it its their fault)
+/// Eg. "1883 - 2021" will use 2021 as the date.
+/// </summary>
+public class LatestPlausibleDateMovieDateTimeResolver : IMovieDateTimeResolver
+{
+    /// <summary>
+    /// Attempts to resolve date and Name from the provided fileName.
+    /// </summary>
+    /// <param name="fileName">Name of video.</param>
+    /// <param name="namingOptions">NamingOptions.</param>
+    /// <returns>null if could not resolve.</returns>
+    public CleanDateTimeResult? Resolve(string fileName, NamingOptions namingOptions)
+    {
+        var regex = new Regex(@"(?'date'\d{4})(?!p)");
```

Comment:
All of these new `Regex` should use [source-generated regex](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators?pivots=dotnet-8-0#source-generation)

Also, change the `\d` to `[0-9]`
> \d matches any decimal digit. It is equivalent to the \p{Nd} regular expression pattern, which includes the standard decimal digits 0-9 as well as the decimal digits of a number of other character sets.